gutenberg.php

```diff
@@ -204,9 +204,11 @@ function gutenberg_pre_init() {
 	// Strip '-src' from the version string. Messes up version_compare().
 	$version = str_replace( '-src', '', $wp_version );
 
-	if ( version_compare( $version, '5.3.0', '<' ) ) {
+	// Compare against major release versions (X.Y) rather than minor (X.Y.Z)
+	// unless a minor release is the actual minimum requirement. WordPress reports
+	// X.Y for its major releases.
+	if ( version_compare( $version, '5.6', '<' ) ) {
 		add_action( 'admin_notices', 'gutenberg_wordpress_version_notice' );
-		return;
```

Comment:
With the `return` removed it will try to load all the files for the plugin even for older versions that aren't supported. Won't it cause an issue?

Reply:
I see an answer:

> After adding an admin notice, the plugin bails with return; and Gutenberg libraries are not loaded. Other hooks are added by the plugin that require those libraries, and a fatal error occurs rather than a compatibility notice. This removes the return, which helps to eliminate immediate fatal errors and allows the notice to be shown.

Should we move all hooks to their own file that gets loaded first in `load.php`?

Reply:
https://github.com/WordPress/gutenberg/pull/30230 should resolve the issue with errors so we can keep `return;` here.